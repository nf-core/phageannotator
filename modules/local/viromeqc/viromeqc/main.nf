process VIROMEQC_VIROMEQC {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-b28a1a551d380ce8d57f9d83894ccb9559b44404:08a4cf815fcef0080ede5b3633202cbda8edf59b-0':
        'biocontainers/mulled-v2-b28a1a551d380ce8d57f9d83894ccb9559b44404:08a4cf815fcef0080ede5b3633202cbda8edf59b-0' }"

    input:
    tuple val(meta), path(reads)
    path(viromeqc_index)

    output:
    tuple val(meta), path("${prefix}.viromeqc.tsv") , emit: enrichment
    path "versions.yml"                             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    viromeQC.py \\
        --input ${reads[0]} ${reads[1]} \\
        --index_dir $viromeqc_index \\
        --output ${prefix}.viromeqc.tsv \\
        --bowtie2_threads $task.cpus \\
        --diamond_threads $task.cpus \\
        --sample_name $prefix \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        viromeqc: \$(echo \$(viromeQC.py --version 2>&1 | sed 's/^.*viromeQC.py../ //; s/Using.*\$//' ))
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.viromeqc.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        viromeqc: \$(echo \$(viromeQC.py --version 2>&1 | sed 's/^.*viromeQC.py //; s/Using.*\$//' ))
    END_VERSIONS
    """
}
