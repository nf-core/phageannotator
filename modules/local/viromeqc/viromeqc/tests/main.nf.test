nextflow_process {

    name "Test Process: VIROMEQC_VIROMEQC"
    script "../main.nf"
    process "VIROMEQC_VIROMEQC"

    // Dependencies
    tag "VIROMEQC_INSTALL"


    test("reads_1.fastq.gz & reads_2.fastq.gz & viromeqc_db") {
        setup {
            run("VIROMEQC_INSTALL") {
                script "../../install/main.nf"
            }
        }

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkifExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkifExists: true),
                    ]
                ]
                input[1] = VIROMEQC_INSTALL.out.viromeqc_index
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("reads_1.fastq.gz & reads_2.fastq.gz & viromeqc_db - stub") {

        options "-stub"

        setup {
            run("VIROMEQC_INSTALL") {
                script "../../install/main.nf"
            }
        }

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkifExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkifExists: true),
                    ]
                ]
                input[1] = VIROMEQC_INSTALL.out.viromeqc_index
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
