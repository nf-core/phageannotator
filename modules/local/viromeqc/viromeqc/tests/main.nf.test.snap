{
    "reads_1.fastq.gz & reads_2.fastq.gz & viromeqc_db": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.viromeqc.tsv:md5,6acbffc97f75774cf9fcc953b98913ed"
                    ]
                ],
                "1": [
                    "versions.yml:md5,1d793f1196d60e1c94969fb0a103e353"
                ],
                "enrichment": [
                    [
                        {
                            "id": "test"
                        },
                        "test.viromeqc.tsv:md5,6acbffc97f75774cf9fcc953b98913ed"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1d793f1196d60e1c94969fb0a103e353"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-21T13:03:00.013402413"
    },
    "reads_1.fastq.gz & reads_2.fastq.gz & viromeqc_db - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.viromeqc.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,2877c7f2bffbd5cd7c2cccc730b9f093"
                ],
                "enrichment": [
                    [
                        {
                            "id": "test"
                        },
                        "test.viromeqc.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2877c7f2bffbd5cd7c2cccc730b9f093"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-21T13:03:23.878748215"
    }
}
